sleap_nn/training/model_trainer.py

```diff
@@ -564,10 +570,21 @@ def __init__(
         if self.model_config.init_weights == "xavier":
             self.model.apply(xavier_init_weights)
 
-        # Pre-trained weights for the encoder stack.
+        # Pre-trained weights for the encoder stack - only for swint and convnext
         if self.model_config.pre_trained_weights:
             self.model.backbone.enc.load_state_dict(ckpt, strict=False)
 
+        # Initializing model (encoder + decoder) with trained ckpts
+        if trained_ckpts_path is not None:
+            print(f"Loading wieghts from `{trained_ckpts_path}` ...")
+            ckpt = torch.load(trained_ckpts_path)
+            ckpt["state_dict"] = {
+                k: ckpt["state_dict"][k]
+                for k in ckpt["state_dict"].keys()
+                if ".head" not in k
```

Comment:
Is this enough of a check?

Are there really no other layers that might have shape mismatches?

Reply:
We need to handle input channel mismatch between the pretrained ckpts and current model. This will be addressed in a separate PR.